h2o-algos/src/main/java/hex/tree/DHistogram.java

```diff
@@ -488,16 +491,22 @@ void updateHisto(double[] ws, double resp[], double[] cs, double[] ys, double[]
       double wy = weight * y;
       double wyy = wy * y;
       int b = bin(col_data);
-      _vals[_vals_dim*b + 0] += weight;
-      _vals[_vals_dim*b + 1] += wy;
-      _vals[_vals_dim*b + 2] += wyy;
+      int binDimStart = _vals_dim*b;
```

Comment:
imho this reads worse than before, maybe if you made binDimStart final it would look better

Reply:
So I made the `binDimStart` final. Maybe this reads worse, however, it is better from a code perspective. But I can change it back if you want @michalkurka.